cpp/src/arrow/flight/sql/odbc/tests/connection_info_test.cc

```diff
@@ -0,0 +1,1453 @@
+
+  this->disconnect();
+}
+
+TYPED_TEST(FlightSQLODBCTestBase, Test_SQL_SEARCH_PATTERN_ESCAPE) {
+  this->connect();
+
+  validate(this->conn, SQL_SEARCH_PATTERN_ESCAPE, L"\\");
+
+  this->disconnect();
+}
+
+TYPED_TEST(FlightSQLODBCTestBase, Test_SQL_SERVER_NAME) {
+  this->connect();
+
+  validateNotEmptySQLWCHAR(this->conn, SQL_SERVER_NAME, false);
+
+  this->disconnect();
+}
+
+TYPED_TEST(FlightSQLODBCTestBase, Test_SQL_STATIC_CURSOR_ATTRIBUTES1) {
+  this->connect();
+
+  validate(this->conn, SQL_STATIC_CURSOR_ATTRIBUTES1, static_cast<SQLUINTEGER>(0));
+
+  this->disconnect();
+}
+
+TYPED_TEST(FlightSQLODBCTestBase, Test_SQL_STATIC_CURSOR_ATTRIBUTES2) {
+  this->connect();
+
+  validate(this->conn, SQL_STATIC_CURSOR_ATTRIBUTES2, static_cast<SQLUINTEGER>(0));
+
+  this->disconnect();
+}
+
+// DBMS Product Information
+
+TYPED_TEST(FlightSQLODBCTestBase, Test_SQL_DATABASE_NAME) {
+  this->connect();
+
+  validate(this->conn, SQL_DATABASE_NAME, L"");
+
+  this->disconnect();
+}
+
+TYPED_TEST(FlightSQLODBCTestBase, Test_SQL_DBMS_NAME) {
+  this->connect();
+
+  validateNotEmptySQLWCHAR(this->conn, SQL_DBMS_NAME, false);
+
+  this->disconnect();
+}
+
+TYPED_TEST(FlightSQLODBCTestBase, Test_SQL_DBMS_VER) {
+  this->connect();
+
+  validateNotEmptySQLWCHAR(this->conn, SQL_DBMS_VER, false);
+
+  this->disconnect();
+}
+
+// Data Source Information
+
+TYPED_TEST(FlightSQLODBCTestBase, Test_SQL_ACCESSIBLE_PROCEDURES) {
+  this->connect();
+
+  validate(conn, SQL_ACCESSIBLE_PROCEDURES, L"N");
+
+  this->disconnect();
+}
+
+TYPED_TEST(FlightSQLODBCTestBase, Test_SQL_ACCESSIBLE_TABLES) {
+  this->connect();
+
+  validate(conn, SQL_ACCESSIBLE_TABLES, L"Y");
+
+  this->disconnect();
+}
+
+TYPED_TEST(FlightSQLODBCTestBase, Test_SQL_BOOKMARK_PERSISTENCE) {
+  this->connect();
+
+  validate(conn, SQL_BOOKMARK_PERSISTENCE, static_cast<SQLUINTEGER>(0));
+
+  this->disconnect();
+}
+
+TYPED_TEST(FlightSQLODBCTestBase, Test_SQL_CATALOG_TERM) {
+  this->connect();
+
+  validate(conn, SQL_CATALOG_TERM, L"");
+
+  this->disconnect();
+}
+
+TYPED_TEST(FlightSQLODBCTestBase, Test_SQL_COLLATION_SEQ) {
+  this->connect();
+
+  validate(conn, SQL_COLLATION_SEQ, L"");
+
+  this->disconnect();
+}
+
+TYPED_TEST(FlightSQLODBCTestBase, Test_SQL_CONCAT_NULL_BEHAVIOR) {
+  this->connect();
+
+  validate(conn, SQL_CONCAT_NULL_BEHAVIOR, static_cast<SQLUSMALLINT>(0));
+
+  this->disconnect();
+}
+
+TYPED_TEST(FlightSQLODBCTestBase, Test_SQL_CURSOR_COMMIT_BEHAVIOR) {
+  this->connect();
+
+  validate(conn, SQL_CURSOR_COMMIT_BEHAVIOR, static_cast<SQLUSMALLINT>(1));
+
+  this->disconnect();
+}
+
+TYPED_TEST(FlightSQLODBCTestBase, Test_SQL_CURSOR_ROLLBACK_BEHAVIOR) {
+  this->connect();
+
+  validate(conn, SQL_CURSOR_ROLLBACK_BEHAVIOR, static_cast<SQLUSMALLINT>(1));
+
+  this->disconnect();
+}
+
+TYPED_TEST(FlightSQLODBCTestBase, Test_SQL_CURSOR_SENSITIVITY) {
+  this->connect();
+
+  validate(conn, SQL_CURSOR_SENSITIVITY, static_cast<SQLUINTEGER>(0));
```

Comment:
Suggest to test against default values provided by the driver.

```suggestion
  validate(conn, SQL_CURSOR_SENSITIVITY, static_cast<SQLUINTEGER>(SQL_UNSPECIFIED));
```